pandas/tests/io/parser/test_na_values.py

```diff
@@ -812,3 +812,23 @@ def test_bool_and_nan_to_float(all_parsers):
     result = parser.read_csv(StringIO(data), dtype="float")
     expected = DataFrame.from_dict({"0": [np.nan, 1.0, 0.0]})
     tm.assert_frame_equal(result, expected)
+
+
+@xfail_pyarrow
+def test_na_values_dict_without_dtype(all_parsers):
+    # GH#59303
+    parser = all_parsers
+    data = """A
+-99
+-99
+-99.0
+-99.0"""
+    # this would FAIL BEFORE this fix
+    result_1 = parser.read_csv(StringIO(data), na_values={"A": [-99.0, -99]})
+    expected_1 = DataFrame.from_dict({"A": [np.nan, np.nan, np.nan, np.nan]})
+    tm.assert_frame_equal(result_1, expected_1)
+
+    # this would PASS even BEFORE this fix
+    result_2 = parser.read_csv(StringIO(data), na_values={"A": [-99, -99.0]})
+    expected_2 = DataFrame.from_dict({"A": [np.nan, np.nan, np.nan, np.nan]})
+    tm.assert_frame_equal(result_2, expected_2)
```

Comment:
Could you use `@pytest.mark.parametrize("values", [[-99.0, -99], [-99, -99.0]])` to de-duplicate this?

Reply:
Implemented as suggested.